### PR TITLE
DOC: Use common variable for serialization note

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

The incremental algorithms all share a common note about what happens when serializing them. This PR moves that note to a common variable, and rewords it a bit along the way.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: daal4py/sklearn/_n_jobs_support.py [hunk 1]
```diff
@@ -214,7 +214,11 @@ def init_with_n_jobs(self, *args, n_jobs=None, **kwargs):
             and "n_jobs : int" not in original_class.__doc__
         ):
             # Python 3.13 removed extra tab in class doc string
-            tab = "    " if sys.version_info.minor < 13 else ""
+            tab = (
+                "    "
+                if (sys.version_info.major == 3 and sys.version_info.minor < 13)
```
Comment (Alexsandruss) [inline daal4py/sklearn/_n_jobs_support.py:219]: Is there a particular reason to explicitly check for the major version? I doubt anyone will ever try to install or build the package in Python 2 env or Python 4 will arrive.

File: sklearnex/_utils.py [hunk 2]
```diff
@@ -155,6 +156,30 @@ def _doc_link_template(self) -> str:
         return f"https://uxlfoundation.github.io/scikit-learn-intelex/latest/non-scikit-algorithms.html#{module_path}.{class_name}"
 
 
+# Adds a small note note about serialization for extension estimators that are incremental.
+# The class docstrings should leave a placeholder '%incremental_serialization_note%' inside
+# their docstrings, which will be replaced by this note.
+def _add_inc_serialization_note(class_docstrings: str) -> str:
```
Comment (Alexsandruss) [inline sklearnex/_utils.py:162]: Docstring gives better readability:
```suggestion
def _add_inc_serialization_note(class_docstrings: str) -> str:
    """Adds a small note note about serialization for extension estimators that are incremental.
    The class docstrings should leave a placeholder '%incremental_serialization_note%' inside
    their docstrings, which will be replaced by this note.
    """
```

File: daal4py/sklearn/_n_jobs_support.py (hunk 1 above)
Comment (david-cortes-intel) [reply daal4py/sklearn/_n_jobs_support.py:219]: It will save us trouble once python4 arrives. Conditions for python2 are irrelevant either way, because there is a minimum version requirement for python.

**Review by Alexsandruss — APPROVED** [review]
